src/resources/ble_esp32.cc

```diff
@@ -880,12 +901,14 @@ void
 BleServiceResource::_on_characteristic_discovered(const BleCallbackScope& scope, const struct ble_gatt_error* error, const struct ble_gatt_chr* chr) {
   switch (error->status) {
     case 0: {
-      auto ble_characteristic =
-          get_or_create_characteristics_resource(
-              chr->uuid, chr->properties, chr->def_handle,
-              chr->val_handle);
-      if (!ble_characteristic) {
-        set_malloc_error(true);
+      for (int i = 0; i < 2; i++) {
```

Comment:
It is weird that this uses a different pattern (looping and retrying here). Why can't `get_or_create_characteristics_resource` do the retry?

If the first attempt fails and we're going to try again on the second iteration of the loop, is it really correct to call `set_malloc_error(true)` and force an OOM?